src/Compilers/CSharp/Portable/Binder/PatternSwitchBinder2.cs

```diff
@@ -0,0 +1,234 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    // We use a subclass of SwitchBinder for the pattern-matching switch statement until we have completed
+    // a totally compatible implementation of switch that also accepts pattern-matching constructs.
+    internal partial class PatternSwitchBinder2 : SwitchBinder
+    {
+        internal PatternSwitchBinder2(Binder next, SwitchStatementSyntax switchSyntax) : base(next, switchSyntax)
+        {
+        }
+
+        /// <summary>
+        /// When pattern-matching is enabled, we use a completely different binder and binding
+        /// strategy for switch statements. Once we have confirmed that it is totally upward
+        /// compatible with the existing syntax and semantics, we will merge them completely.
+        /// However, until we have edit-and-continue working, we continue using the old binder
+        /// when we can.
+        /// </summary>
+        private bool UseV8SwitchBinder
+        {
+            get
+            {
+                    var parseOptions = SwitchSyntax?.SyntaxTree?.Options as CSharpParseOptions;
+                    return
```

Comment:
>return [](start = 20, length = 6)

Alignment looks off

Reply:
It appears to be fixed in later PRs for the feature.

---
In reply to: [156523146](https://github.com/dotnet/roslyn/pull/23209#discussion_r156523146) [](ancestors = 156523146)